syntax = "proto3";

option java_package = "com.is.common";

package is.common;

message SamplingSettings {
  oneof rate {
    float frequency = 1; // Sampling frequency in hertz
    float period = 2; // Sampling rate period in seconds
  }
  // add delay...
}

// Used to select the desired fields of a message on a "Get" rpc
message FieldSelector {
  repeated uint32 fields = 1;
}

//
enum StatusCode {
  UNKNOWN = 0;
  OK = 1;
  CANCELLED = 2;
  INVALID_ARGUMENT = 3;
  DEADLINE_EXCEEDED = 4;
  NOT_FOUND = 5;
  ALREADY_EXISTS = 6;
  PERMISSION_DENIED = 7;
  UNAUTHENTICATED = 8;
  FAILED_PRECONDITION = 9;
  OUT_OF_RANGE = 10;
  UNIMPLEMENTED = 11;
  INTERNAL_ERROR = 12;
}

message Status {
  StatusCode code = 1;
  // Human readable message explaining the error
  string why = 2; 
}

message DoubleMatrix {
  uint32 rows = 1;
  uint32 cols = 2;
  repeated double data = 3 [packed=true];
}